src/test/java/brickbreakertest/GameTest.java

```diff
@@ -1,29 +1,49 @@
 package brickbreakertest;
 
 import brickbreaker.model.*;
+import brickbreaker.view.GameComponent;
+import brickbreaker.view.GameFrame;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
 
 public class GameTest {
 
     //test for ball.moveBall() straight
-    @Test
-    public void moveBall() {
-        Ball ball = new Ball(10, 10, 15);
-        ball.moveBall();
-      //  assertEquals()
-
-    }
+    //    @Test
+    //    public void moveBall() {
+    //
+    //      //  assertEquals()
+    //
+    //    }
 
     //test for ball.reflectOffWall()
+    @Test
     public void reflectOffWall() {
+        Game game = mock();
+        Ball ball = mock();
+        doReturn(ball).when(game).getBall();
+        //when
+        game.ballHitWall(anyInt(), anyInt());
+        ball.reflectOffWall(anyInt(), anyInt());
 
+        //then
+        verify(game).ballHitWall(anyInt(), anyInt());
+        verify(ball).reflectOffWall(anyInt(), anyInt());
```

Comment:
You didn't really understand what's going on here. 

You aren't testing anything. 

You are calling methods on a mock and then verifying that those methods have been called

Reply:
Could we please have some sort of tutorial on mocks tomorrow, because I don't really understand how to use them still.

Reply:
Sure.